slf4j-api/src/main/java/org/slf4j/Logger.java

```diff
@@ -172,6 +182,16 @@ default public LoggingEventBuilder makeLoggingEventBuilder(Level level) {
      */
     public void trace(String msg, Throwable t);
 
+    /**
+     * Log an exception (throwable) at the TRACE level with an
+     * accompanying message.
+     *
+     * @param msgSupplier supplier for the message accompanying the exception
+     * @param t   the exception (throwable) to log
+     * @since 1.4
+     */
+    public void trace(Supplier<String> msgSupplier, Throwable t);
```

Comment:
Better to write
`public void trace(Throwable t, Supplier<String> msgSupplier);`
In kotlin after this we can write `logger.trace(exception) { "Message" }`

Reply:
@alekseytomin This would be counter-intuitive for Java programmers though, as in all other cases the throwable is the last parameter.

Reply:
Ok. We can write kotlin extensions.